spec/src/main/asciidoc/ecosystem.asciidoc

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2018 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+//
+
+= MicroProfile Config and Service Mesh
+
+MicroProfile Config provides a solution to externalise the configuration. The default config sources includes environment variables, system properties and microprofile-config.properties on the classpath. 
+
+The properties defined in Kubernetes config map can be transformed to environment variables via envFrom. After a cloud-native microservice is deployed, if an extra config source is needed, this file can be mounted to configmap, which then use envFrom facility to expose the properties to the microservie. 
+
+[source, text]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: example-config
+  namespace: default
+data:
+  example.property.1: hello
+  example.property.2: world
+  
+   
+----
+----
+# Use envFrom to load ConfigMaps into environment variables
+
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: mans-not-hot
+  labels:
+    app: mans-not-hot
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: mans-not-hot
+  template:
+    metadata:
+      labels:
+        app: mans-not-hot
+    spec:
+      containers:
+        - name: app
+          image: gcr.io/mans-not-hot/app:bed1f9d4
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 80
+          envFrom:
+          - configMapRef:
+              name: example-config
+----
+
+Therefore, config properties specified in the config map are automatically injectable to the microservices via MicroProfile Config APIs.
+
+[source, java]
+----
+@ApplicationScoped
```

Comment:
I'm not in favor of having usages or sub spec details included in this document.

It's rife for becoming out of date too quickly. I would prefer to see links to other spec docs as appropriate.

Reply:
deleted